docs/adapters/configuration.md

```diff
@@ -0,0 +1,110 @@
+## Goal
```

Comment:
Cool - I moved the design doc to the wiki: https://github.com/juttle/juttle/wiki/adapter-config-design